lib/ws_servers/api/send_authenticated.js

```diff
@@ -28,7 +27,7 @@ module.exports = async (ws, db, d, opts) => {
     .filter(c => c.cid !== CREDENTIALS_CID)
 
   // Attempt to decrypt & open connections for all valid credentials
-  await PI.forEach(apiCredentials, async (credentials) => {
+  for (const credentials of apiCredentials) {
```

Comment:
very nice!!

in general this belongs in a separate PR, has not much to do with algo orders?

Reply:
PS: as mentioned in slack i had to extend and change this section as part of the paper trading mode feature.

see my chat message that i refactor pi-iteration out and this PR

Reply:
can you revert?

Reply:
yeah .. i had this change before i saw your review .. so forgot .. i will revert this